src/tools/rbd/action/DiskUsage.cc

```diff
@@ -152,6 +153,7 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
       continue;
     }
 
+    bool found_from_snap = (from_snapname == nullptr);
```

Comment:
Avoiding the need to test "from_snapname == nullptr" two times below by just combing it into the single bool -- if we aren't restricting the starting snapshot, might as well treat it as we've found the starting snapshot.